velox/docs/functions/presto/decimal.rst

```diff
@@ -0,0 +1,201 @@
+=================
+Decimal Operators
+=================
+
+DECIMAL type is designed to represent floating point numbers precisely.
+Mathematical operations on decimal values are exact, except for division. On
+the other hand, DOUBLE and REAL types are designed to represent floating point
+numbers approximately. Mathematical operations on double and real values are
+approximate.
+
+For example, the number 5,000,000,000,000,000 can be represented using DOUBLE.
+However, the number 5,000,000,000,000,000.15 cannot be represented using
+DOUBLE, but it can be represented using DECIMAL. See
+https://en.wikipedia.org/wiki/Double-precision_floating-point_format for more
+details.
+
+DECIMAL type has 2 parameters: precision and scale. Precision is the total
+number of digits used to represent the number. Scale is the number of digits
+after the decimal point. Naturally, scale must not exceed precision. In
+addition, precision cannot exceed 38.
+
+::
+
+	decimal(p, s)
+
+	p >= 1 && p <= 38
+	s >= 0 && s <= p
+
+The number 5,000,000,000,000,000.15 can be represented using DECIMAL(18, 2).
+This number needs at least 18 digits (precision) of which at least 2 digits
+must appear after the decimal point (scale). This number can be represented
+using any DECIMAL type where scale >= 2 and precision is >= scale + 16.
+
+Addition and Subtraction
+------------------------
+
+To represent the results of adding two decimal numbers we need to use max
+(s1, s2) digits after the decimal point and max(p1 - s1, p2 - s2) + 1 digits
+before the decimal point.
+
+::
+
+	p = max(p1 - s1, p2 - s2) + 1 + max(s1, s2)
+	s = max(s1, s2)
+
+It is easiest to understand this formula by thinking about column addition where
+we place two numbers one under the other and line up decimal points.
+
+::
+
+        1.001
+     9999.5
+   -----------
+    10000.501
+
+We can see that the result needs max(s1, s2) digits after the decimal point and
+max(p1 - s1, p2 - s2) + 1 digits before the decimal point.
+
+The precision of the result may exceed 38. There are two options. One option is
+to say that addition and subtraction is supported as long as p <= 38 and reject
+operations that produce p > 38. Another option is to cap p at 38 and allow the
+operation to succeed as long as the actual result can be represented using 38
+digits. In this case, users experience runtime errors when the actual result
+cannot be represented using 38 digits. Presto implements the second option.
```

Comment:
Should this say `Velox implements the second option.` ?

Reply:
Velox implements Presto's behavior. How about if I change this to 

```
Presto implements the second option and Velox matches it.
```

Reply:
These operations for Spark may have a different behavior (both are implemented by Velox though).

Reply:
Since this document is under `functions/presto`, the Presto context is implicit in my opinion.
`Presto implements the second option and Velox matches it.` is reasonable to me. Thanks!